bin/prover-client/src/checkpoint_runner/fetch.rs

```diff
@@ -0,0 +1,21 @@
+use jsonrpsee::http_client::HttpClient;
+use strata_rpc_api::StrataApiClient;
+use tracing::error;
+
+use super::errors::CheckpointResult;
+use crate::checkpoint_runner::errors::CheckpointError;
+
+/// Fetches the latest checkpoint index from the sequencer client.
+pub async fn fetch_latest_checkpoint_index(cl_client: &HttpClient) -> CheckpointResult<u64> {
+    cl_client
+        .get_latest_checkpoint_index(None)
+        .await
+        .map_err(|e| {
+            error!("Failed to fetch current checkpoint index: {e}");
```

Comment:
Use log params instead of interpolation: `error!(err = ?e, "failed to fetch current checkpoint index");`